test/test_datasets.py

```diff
@@ -29,6 +41,174 @@ def test_get_intersection_range(self):
         s2 = pd.Series(pd.date_range('2020-10-01', '2020-12-01'))
         self._test_get_intersection_range(s1, s2, 32)
 
+    @patch('pyronear_ds.datasets.utils.requests.get')
+    def test_url_retrieve(self, mock_get):
+
+        mock_get.return_value.status_code = 200
+        mock_get.return_value.content = bytes("WEATHER OR WILDFIRE FILE", 'utf-8')
+        content = utils.url_retrieve('url')
+        self.assertIsInstance(content, bytes)
+
+        mock_get.return_value.status_code = 400
+        mock_get.return_value.content = bytes("WEATHER OR WILDFIRE FILE", 'utf-8')
+        self.assertRaises(requests.exceptions.ConnectionError, utils.url_retrieve, 'url')
+
+    def test_get_fname(self):
+
+        url_firms = "https://firms.modaps.eosdis.nasa.gov/data/active_fire/c6/csv/MODIS_C6_Europe_24h.csv"
+        url_ghcn = "https://www1.ncdc.noaa.gov/pub/data/ghcn/daily/by_year/2020.csv.gz"
+        url_isd = "https://www.ncei.noaa.gov/data/global-hourly/archive/csv/2020.tar.gz"
+
+        self.assertEqual(utils.get_fname(url_firms), ("MODIS_C6_Europe_24h", "csv", None))
+        self.assertEqual(utils.get_fname(url_ghcn), ("2020", "csv", "gz"))
+        self.assertEqual(utils.get_fname(url_isd), ("2020", None, "tar.gz"))
+
+    @staticmethod
+    def _fake_csv(destination, fname):
```

Comment:
small naming suggestion : `def _mock_csv` since mock is more neutral than "fake" :sweat_smile: